src/goal.rs

```diff
@@ -953,37 +867,10 @@ impl<'a> Goal<'a> {
       // Remove old variable from the egraph and context
       remove_node(&mut new_goal.egraph, &SymbolLang::leaf(var));
       // warn!("removing var {}", var);
-      // FIXME: is this OK? add a full_context?
```

Comment:
**comment**: So it's OK to have the context track all past variables? I noticed that the comment for it indicates that it will have old variables in it.

IIRC I thought it was OK because the way it was used, it was fine for the `local_context` to have old vars.

Reply:
Yes, it's okay!